### PR TITLE
Format generated source code before writing it to the destination

### PR DESCRIPTION
**Pull request by peterstace**

### CODE REVIEW

File: mockgen/mockgen.go [hunk 1]
```diff
@@ -86,9 +88,7 @@ func main() {
 		packageName = "mock_" + sanitize(pkg.Name)
 	}
 
-	g := generator{
-		w: dst,
-	}
+	g := generator{}
```
Comment (dsymonds) [inline mockgen/mockgen.go]: let's make this

```go
g := new(generator)
```

File: mockgen/mockgen.go [hunk 2]
```diff
@@ -411,3 +414,15 @@ func (g *generator) GenerateMockRecorderMethod(mockType string, m *model.Method)
 	g.p("}")
 	return nil
 }
+
+// Format gives a go-formatted copy of the buffer.
```
Comment (dsymonds) [inline mockgen/mockgen.go]: let's call this Output.

// Output returns the generator's output, formatted in the standard Go style.

File: mockgen/mockgen.go [hunk 3]
```diff
@@ -411,3 +414,15 @@ func (g *generator) GenerateMockRecorderMethod(mockType string, m *model.Method)
...
+func (g *generator) Format() []byte {
+	src, err := format.Source(g.buf.Bytes())
+	if err != nil {
+		// This should never happen, but if it does, give the user a warning
```
Comment (dsymonds) [inline mockgen/mockgen.go]: If it should never happen, log.Fatalf is good enough.

**Comment by dsymonds** [comment]
This looks fine. Just a few nits.

**Comment by peterstace** [comment]
I've signed the CLA.

**Comment by dsymonds** [comment]
Merged in d581abf.

**Pull request closed without merging**